### PR TITLE
Added load testing scripts

### PR DESCRIPTION
**Pull request by Jineshdarjee**
## Description:
1. Added the load testing scripts for Arka and Skandha.

## Newly Created Files and Modified Files:
* **New File:** test/specs/mainnet-api/arka/LoadTesting.spec.js
* **New File:** test/specs/mainnet-api/skandha/LoadTesting.spec.js
* **New File:** test/utils/Arka_CsvToDatabase.js
* **New File:** test/utils/Skandha_CsvToDatabase.js
* **Modified File:** package.json
* **Modified File:** .env

### CODE REVIEW

**Comment by Jineshdarjee** [comment]
## Description:
Modified Skandha test cases.

## Newly Created Files and Modified Files:
- **Modified File:** package.json
- **Modified File:** test/specs/mainnet-api/skandha/goerli.spec.js
- **Modified File:** test/specs/mainnet-api/skandha/mumbai.spec.js

**Review by IAmKio — CHANGES_REQUESTED** [review]
Left a few comments that need to be addressed. Please remember that anything you write needs to be written in a universal way and not for your local machine.

File: .env [hunk 1]
```diff
@@ -15,4 +15,8 @@ PROJECT_KEY=_project_key_
 PROJECT_KEY_TESTNET=_project_key_testnet_
 API_KEY=_api_key_
 INCORRECT_API_KEY=_incorrect_api_key_
-INVALID_API_KEY=_invalid_api_key_
\ No newline at end of file
+INVALID_API_KEY=_invalid_api_key_
+DB_HOST=_db_host_
+DB_USER=_db_user_
+DB_PASSWORD=_db_password_
+DB_DATABASE=_bd_database_
```
[the remark this answers is not in the document]
Comment (Jineshdarjee) [reply .env:22]: Modified the code

File: test/specs/mainnet-api/arka/LoadTesting.spec.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import saveToDatabase from '../../../utils/Arka_CsvToDatabase.js';
+
+describe('CSV to Database Test', function () {
+  it('Saving the Arka CSV data to the database', function (done) {
+    const csvFilePath =
+      'C:/Users/admin/apache-jmeter-5.6.2/apache-jmeter-5.6.2/arka_summaryreport.csv';
```
[the remark this answers is not in the document]
Comment (Jineshdarjee) [reply test/specs/mainnet-api/arka/LoadTesting.spec.js:6]: We are performing the load testing from the local system using the JMeter tool. So paths of the file are saved in the local system only.
But now I have added both paths in the .env file.

File: test/specs/mainnet-api/skandha/LoadTesting.spec.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import saveToDatabase from '../../../utils/Skandha_CsvToDatabase.js';
+
+describe('CSV to Database Test', function () {
+  it('Saving the Skandha CSV data to the database', function (done) {
+    const csvFilePath =
+      'C:/Users/admin/apache-jmeter-5.6.2/apache-jmeter-5.6.2/skandha_summaryreport.csv';
```
[the remark this answers is not in the document]
Comment (Jineshdarjee) [reply test/specs/mainnet-api/skandha/LoadTesting.spec.js:6]: We are performing the load testing from the local system using the JMeter tool. So paths of the file are saved in the local system only.
But now I have added both paths in the .env file.

File: test/specs/mainnet-api/skandha/goerli.spec.js [hunk 4]
```diff
@@ -1,12 +1,18 @@
 import { assert } from 'chai';
 import data from '../../../data/api_testData.json' assert { type: 'json' };
+import customRetryAsync from '../../../utils/baseTest.js';
+// import Helper from '../../../utils/Helper.js';
 import addContext from 'mochawesome/addContext.js';
+import { PrimeSdk } from '@etherspot/prime-sdk';
+import { ethers } from 'ethers';
+import data1 from '../../../data/testData.json' assert { type: 'json' };
 import * as dotenv from 'dotenv';
 dotenv.config(); // init dotenv
```
[the remark this answers is not in the document]
Comment (Jineshdarjee) [reply test/specs/mainnet-api/skandha/goerli.spec.js:10]: Updated the code from all the test case files.

File: test/specs/mainnet-api/skandha/goerli.spec.js [hunk 5]
```diff
@@ -1,12 +1,18 @@
 import { assert } from 'chai';
 import data from '../../../data/api_testData.json' assert { type: 'json' };
+import customRetryAsync from '../../../utils/baseTest.js';
+// import Helper from '../../../utils/Helper.js';
 import addContext from 'mochawesome/addContext.js';
+import { PrimeSdk } from '@etherspot/prime-sdk';
+import { ethers } from 'ethers';
+import data1 from '../../../data/testData.json' assert { type: 'json' };
 import * as dotenv from 'dotenv';
```
[the remark this answers is not in the document]
Comment (Jineshdarjee) [reply test/specs/mainnet-api/skandha/goerli.spec.js:9]: Updated the code from all the test case files.

File: test/specs/mainnet-api/skandha/goerli.spec.js [hunk 6]
```diff
@@ -280,4 +286,245 @@ describe('Performance testing of Skandha Endpoints with Mumbai Network', functio
...
+
+      // get EtherspotWallet address
+      try {
+        await goerliTestNetSdk.getCounterFactualAddress();
+      } catch (e) {
+        console.error(e);
+        const eString = e.toString();
+        addContext(test, eString);
+        assert.fail(
+          'The Etherspot Wallet Address is not displayed successfully.',
```
[the remark this answers is not in the document]
Comment (Jineshdarjee) [reply test/specs/mainnet-api/skandha/goerli.spec.js:319]: assert.fail accept a single string parameter only.
and however, all the console errors are captured in mochawesome report so we can track errors from the report.

File: test/specs/mainnet-api/skandha/goerli.spec.js [hunk 7]
```diff
@@ -280,4 +286,245 @@ describe('Performance testing of Skandha Endpoints with Mumbai Network', functio
...
+      }
+
+      // clear the transaction batch
+      try {
+        await goerliTestNetSdk.clearUserOpsFromBatch();
+      } catch (e) {
+        console.error(e);
+        const eString = e.toString();
+        addContext(test, eString);
+        assert.fail('The transaction of the batch is not clear correctly.');
```
[the remark this answers is not in the document]
Comment (Jineshdarjee) [reply test/specs/mainnet-api/skandha/goerli.spec.js:330]: assert.fail accept a single string parameter only.
and however, all the console errors are captured in mochawesome report so we can track errors from the report.

File: test/specs/mainnet-api/skandha/goerli.spec.js [hunk 8]
```diff
@@ -280,4 +286,245 @@ describe('Performance testing of Skandha Endpoints with Mumbai Network', functio
...
+        await goerliTestNetSdk.addUserOpsToBatch({
+          to: data1.recipient,
+          value: ethers.utils.parseEther(data1.value),
+        });
+      } catch (e) {
+        console.error(e);
+        const eString = e.toString();
+        addContext(test, eString);
+        assert.fail(
+          'The addition of transaction in the batch is not performed.',
```
[the remark this answers is not in the document]
Comment (Jineshdarjee) [reply test/specs/mainnet-api/skandha/goerli.spec.js:344]: assert.fail accept a single string parameter only.
and however, all the console errors are captured in mochawesome report so we can track errors from the report.

File: test/specs/mainnet-api/skandha/goerli.spec.js [hunk 9]
```diff
@@ -280,4 +286,245 @@ describe('Performance testing of Skandha Endpoints with Mumbai Network', functio
...
+        // Nonce = op.nonce;
+        // InitCode = op.initCode;
+        // CallData = op.callData;
+        // CallGasLimit = op.callGasLimit;
+        // VerificationGasLimit = op.verificationGasLimit;
+        // MaxFeePerGas = op.maxFeePerGas;
+        // MaxPriorityFeePerGas = op.maxPriorityFeePerGas;
+        // PaymasterAndData = op.paymasterAndData;
+        // PreVerificationGas = op.preVerificationGas;
+        // Signature = op.signature;
```
[the remark this answers is not in the document]
Comment (Jineshdarjee) [reply test/specs/mainnet-api/skandha/goerli.spec.js:377]: Removed the code.

File: test/specs/mainnet-api/skandha/goerli.spec.js [hunk 10]
```diff
@@ -280,4 +286,245 @@ describe('Performance testing of Skandha Endpoints with Mumbai Network', functio
...
+      } catch (e) {
+        console.error(e);
+        const eString = e.toString();
+        addContext(test, eString);
+        assert.fail(
+          'The estimate transactions added to the batch and get the fee data for the UserOp is not performed.',
```
[the remark this answers is not in the document]
Comment (Jineshdarjee) [reply test/specs/mainnet-api/skandha/goerli.spec.js:383]: assert.fail accept a single string parameter only.
and however, all the console errors are captured in mochawesome report so we can track errors from the report.

File: test/specs/mainnet-api/skandha/goerli.spec.js [hunk 11]
```diff
@@ -280,4 +286,245 @@ describe('Performance testing of Skandha Endpoints with Mumbai Network', functio
...
+            params: [op, data.entryPointAddress],
+          }),
+        });
+        if (!response.ok) {
+          console.error('Response status:', response.status);
+          addContext(test, 'Response status: ' + response.status);
+          const errorResponse = await response.text();
+          console.error('Error response:', errorResponse);
+          addContext(test, 'Error response:' + errorResponse);
+          assert.fail('Getting an error');
```
[the remark this answers is not in the document]
Comment (Jineshdarjee) [reply test/specs/mainnet-api/skandha/goerli.spec.js:406]: assert.fail accept a single string parameter only.
and however, all the console errors are captured in mochawesome report so we can track errors from the report.

File: test/specs/mainnet-api/skandha/goerli.spec.js [hunk 12]
```diff
@@ -280,4 +286,245 @@ describe('Performance testing of Skandha Endpoints with Mumbai Network', functio
...
+          addContext(test, 'Time to First Byte (TTFB): ' + ttfb_s + ' second');
+
+          const returnedValue = await response.json();
+          const errorMessage = returnedValue.error.message;
+          if (errorMessage.includes('invalid signature length')) {
+            console.log('Value returned:', returnedValue);
+            const returnedValueString = JSON.stringify(returnedValue);
+            addContext(test, 'Value returned: ' + returnedValueString);
+          } else {
+            assert.fail('The response is not getting correct.');
```
[the remark this answers is not in the document]
Comment (Jineshdarjee) [reply test/specs/mainnet-api/skandha/goerli.spec.js:422]: assert.fail accept a single string parameter only.
and however, all the console errors are captured in mochawesome report so we can track errors from the report.

File: test/specs/mainnet-api/skandha/goerli.spec.js [hunk 13]
```diff
@@ -280,4 +286,245 @@ describe('Performance testing of Skandha Endpoints with Mumbai Network', functio
...
+          }
+        }
+      } catch (e) {
+        console.error('Fetch error:', e);
+        const eString = e.toString();
+        addContext(test, eString);
+        assert.fail('Getting an error');
```
[the remark this answers is not in the document]
Comment (Jineshdarjee) [reply test/specs/mainnet-api/skandha/goerli.spec.js:429]: assert.fail accept a single string parameter only.
and however, all the console errors are captured in mochawesome report so we can track errors from the report.

File: test/utils/Arka_CsvToDatabase.js [hunk 14]
```diff
@@ -0,0 +1,51 @@
+import mysql from 'mysql2';
+import fs from 'fs';
+import csv from 'csv-parser';
+import * as dotenv from 'dotenv';
+dotenv.config(); // init dotenv
```
[the remark this answers is not in the document]
Comment (Jineshdarjee) [reply test/utils/Arka_CsvToDatabase.js:5]: Updated the code

File: test/utils/Arka_CsvToDatabase.js
[the remark this answers is not in the document]
Comment (Jineshdarjee) [reply test/utils/Arka_CsvToDatabase.js:1]: **Current Load Testing Flow:**
* Perform the load testing on JMeter in the local system.
* The CSV file is created/updated in the local system.
* Run the test case of load testing, So CSV File details will save in the database.

**Review by ch4r10t33r — CHANGES_REQUESTED** [review]
Filenames should be capitalised and the directories should/shouldn't contain "_".  Can we pls be consistent with naming conventions of files?

**Pull request closed without merging**